prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -1048,12 +1056,22 @@ void rxr_pkt_handle_recv_completion(struct rxr_ep *ep,
 #endif
 	peer = rxr_ep_get_peer(ep, pkt_entry->addr);
 	assert(peer);
+	if (peer->is_local && lower_ep_type == EFA_EP) {
+		/*
+		 * This happens when the peer is on same instance, but chose to
+		 * use EFA device to communicate with me. In this case, we respect
+		 * that and will not use shm with the peer.
+		 * TODO: decide whether to use shm through handshake packet.
+		 */
+		peer->is_local = 0;
```

Comment:
Is this modification still necessary after we are using `ep->use_shm_for_tx` to determine whether `shm` is used?